ITISFoundation/osparc-simcore
# 🐛 Fix/auth resend during registration

## What do these changes do?

During the pre-demo we could not resend 2FA code during  registration  (failed with ``Unauthorized``) but we could resend during login.

- 🐛 Authorizes to ``auth_resend_2fa_code`` from ``auth_login`` **and ``auth_register_phone``** (<5 requests/session)
- ✨ ``auth_register_phone``and ``auth_login`` both return ``data.parameters.retry_2fa_after`` in seconds.
- ✨  Adds human-readable error message  for ``Unauthorized`` error (see new ``MSG_UNAUTHORIZED_...``
- ✨ Adjusted number of retrials (i.e. number of continues requests allowed per session w/o restarting workflow)
   -  ``auth_register_phone`` x1
   -  ``auth_resend_2fa_code`` x5
   - ``auth_login_2fa`` (i.e. submit code) x3 
- ♻️ Changes some handle names (for clarity)


## Related issue/s

- Fixes bug in ITISFoundation/osparc-issues#765


## How to test

1. used invitation
2. registered (user+password+confirmation)
3. received link -> click -> email confirmed -> redirects to login page
4. login page: user+password
5. register phone
6. resend SMS
7. resend via email
8. submit (wrong) code
9. if unauth above, i can restart browser and go to 4 (perhaps we can have a button??)
10. if code is ok, we are logged in

@odeimaiz. I noticed:
  - count-down is submit via Email only happens the first time I resend 
  - should add a way to go back or even restart login/registration (see 9)
  - send ``null`` code returns 422 but is not shown nicely in the UI
  - how do you read response messages? ``reason`` or ``message``??

## Checklist

- [x] add extra test with other workflows
- [x] ``make version-*``
- [x] ``make openapi.json``
